testdata/nil-len.go

```diff
@@ -35,4 +35,6 @@ func fn() {
 
 	if s != nil && len(s) == 0 { // nil check is not redundant here
 	}
+	if s != nil && len(s) != 1 { // nil check is not redundant here
```

Comment:
`s != nil && len(s) >[=] X` is also redundant. As well as `s != nil && len(s) != X` where X is not 0.

`s != nil && len(s) < X` however is not redundant. E.g. for `s = nil`:
- `s != nil && len(s) < 5` is false
- `len(s) < 5` is true